src/main/java/com/gooddata/executeafm/afm/SimpleMeasureDefinition.java

```diff
@@ -73,19 +89,30 @@ public SimpleMeasureDefinition(final ObjQualifier item, final Aggregation aggreg
 
     /**
      * Creates new definition
-     * @param item item which is measured, can be attribute, fact or another measure
-     * @param aggregation additional aggregation applied
-     * @param computeRatio whether should be shown as ratio
-     * @param filters additional filters applied
+     *
+     * @param item
+     *         item which is measured, can be attribute, fact or another measure
+     * @param aggregation
+     *         additional aggregation applied
+     * @param computeRatio
+     *         whether should be shown as ratio
+     * @param filters
+     *         additional filters applied
      */
-    public SimpleMeasureDefinition(final ObjQualifier item, final Aggregation aggregation, final Boolean computeRatio,
-                                   final FilterItem... filters) {
+    public SimpleMeasureDefinition(final ObjQualifier item, final Aggregation aggregation, final Boolean computeRatio, final FilterItem... filters) {
         this(item, aggregation, computeRatio, asList(filters));
     }
 
     @Override
-    public MeasureDefinition withObjUriQualifier(final UriObjQualifier qualifier) {
-        return new SimpleMeasureDefinition(qualifier, aggregation, computeRatio, filters);
+    public MeasureDefinition withObjUriQualifiers(final ObjQualifierConverter objQualifierConverter) {
+        if (item instanceof IdentifierObjQualifier) {
+            final IdentifierObjQualifier identifierObjQualifier = (IdentifierObjQualifier) item;
```

Comment:
again the same piece of code - let's introduce some utility function

Reply:
Fixed via `ObjIdentifierUtilities`.